### PR TITLE
MSCCLPP integration

### PR DESCRIPTION
**Pull request by Binyang2014**
MSCCLPP integration

### CODE REVIEW

File: examples/mscclang/allreduce_a100_allpairs.py
Comment (yzygitzh) [inline examples/mscclang/allreduce_a100_allpairs.py:1]: Seems only whitespace changes, maybe drop the changes in this file to make PR simpler?

File: msccl/language/rank_dag.py [hunk 1]
```diff
@@ -172,11 +305,236 @@ def convert_set_list(self):
...
+                                merge_op(op, seq_op)
+                                tb.ops.remove(seq_op)
+                                queue.remove(seq_op)
+                                fused = True
+                        if fused:
+                            continue
+                    queue = queue[1:]
+
+    def optimize_mscclpp(self):
+        self._optimize_redandant_signal_wait()
```
Comment (yzygitzh) [inline msccl/language/rank_dag.py:532]: redandant -> redundant

**Comment by Binyang2014** [comment]
Drop this PR and move to new repo

**Comment by lerrorgk** [comment]
> Drop this PR and move to new repo

Is new repo https://github.com/Azure/msccl-tools ?

**Pull request closed without merging**